frontend/packages/llama-stack-modular-ui/bff/internal/api/models_handler_test.go

```diff
@@ -0,0 +1,294 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/opendatahub-io/llama-stack-modular-ui/internal/config"
+	"github.com/opendatahub-io/llama-stack-modular-ui/internal/constants"
+	"github.com/opendatahub-io/llama-stack-modular-ui/internal/integrations"
+	"github.com/opendatahub-io/llama-stack-modular-ui/internal/mocks"
+	"github.com/opendatahub-io/llama-stack-modular-ui/internal/models"
+	"github.com/opendatahub-io/llama-stack-modular-ui/internal/repositories"
+	"github.com/stretchr/testify/assert"
+)
+
+func createTestApp() *App {
+	mockClient, _ := mocks.NewLlamastackClientMock()
+	return &App{
+		config: config.EnvConfig{
+			Port: 8080,
+		},
+		repositories: repositories.NewRepositories(mockClient),
+	}
+}
+
+func TestGetAllModelsHandlerSuccess(t *testing.T) {
+	app := createTestApp()
```

Comment:
Even for all the handler scenarios - you can refactor this into Table format (example in the previous comment) so that you need worry about setting up clients multiple times.